lesson_2/calculator.py

```diff
@@ -1,16 +1,44 @@
-# calculator program that will:
-    # Ask the user for two numbers.
-    # Ask the user for operation to perform: add, subtract, multiply or divide.
-    # Perform the calculation and display the results
+'''calculator program that will:
+    Ask the user for two numbers.
+    Ask the user for operation to perform: add, subtract, multiply or divide.
+    Perform the calculation and display the results'''
 
-print('Welcome to Calculator!')
+def prompt(message):
+    print(f"==> {message}")
 
-# no error for input/ will crash if input is wrong --> would add error if was function
-num1 = int(input('Enter a number: '))
-num2 = int(input('Enter another number: '))
+def invalid_input(str_num):
+    try:
+        int(str_num)
+    except ValueError:
+        return True
+    return False
 
-operation = input('What operation would you like to perform?\n'
-                  '1) Addition 2) Subtraction 3) Multiplication 4) Division: ')
+prompt('Welcome to Calculator!')
+
+# NUM1
+prompt('Enter a number!')
+num1 = input()
+while invalid_input(num1):
+    prompt("Error: That isn't a valid number, expecting an integer!")
+    num1 = input()
```

Comment:
suggestion: make this a function so you can reuse it for num2

Anytime you find yourself copy/pasting code or re-writing the same code you should reach for a function. It will clean up this code and make it more readable.

Reply:
too many moving parts and going down a rabbit hole, so it's noted but I wont be adding the function @Doxify